rskj-core/src/main/java/org/ethereum/core/Transaction.java

```diff
@@ -277,8 +491,20 @@ public Keccak256 getRawHash() {
         return this.rawHash;
     }
 
-    public byte[] getNonce() {
-        return nullToZeroArray(nonce);
+    public Keccak256 getRawHash(int signerIndex) {
+        byte[] plainMsg = this.getEncodedRaw(signerIndex);
+        return new Keccak256(HashUtil.keccak256(plainMsg));
+    }
+
+    public byte[] getSingleNonce() {
+        if (nonce.size() > 1) {
+            throw new UnsupportedOperationException("Tried get only one nonce");
+        }
+        return nonce.isEmpty() ? EMPTY_BYTE_ARRAY : nonce.get(0);
+    }
+
+    public List<byte[]> getNonces() {
```

Comment:
This method seems useless for me. I'd rather have a method that returns a map (Sender,nonce) than just the nonces.

Reply:
I suppose it could still exists for testing.